src/neo-vm/ExecutionEngine.cs

```diff
@@ -61,9 +61,14 @@ internal protected set
             }
         }
 
-        public ExecutionEngine()
+        public ExecutionEngine() : this(new ReferenceCounter())
         {
-            ResultStack = new EvaluationStack(ReferenceCounter);
+        }
+
+        protected ExecutionEngine(ReferenceCounter referenceCounter)
+        {
+            this.ReferenceCounter = referenceCounter;
```

Comment:
So, my previous question comes here: do we need to ensure that "globally" a nesting access (call X -> call Y -> call Z -> ...) in such a way that it surpasses limits for array counts, etc, in a dangerous way?

Reply:
What do you mean? This change is to ensure that it won't surpass the limits.